### ISSUE TITLE
Expand current file's directory

### ISSUE THREAD

**Issue by ThermalSpan** [benekastah/neomake#155]
First, I apologize If I'm doing this incorrectly, I'm very new to collaborating on software. 

I wasn't able to find a way to add the current file's directory to the argument list for a maker definition. I understand the functionality provided by the using 'Neomake! <maker>' but if I'm working on two different files in different directory's then using the current directory doesn't seem practical. 

Basically, would it be possible to add something like '%:p' in a maker definition that expands to the file's directory?

For some background, I'm want to write a maker for a java <a href="https://processing.org"> processing </a> sketch. The required command looks something like 
```
processing-java --run --force --sketch="$CURDIRECTORY" --output="$CURDIRECTORY/output"
```

**Comment by Neki** [comment]
Placeholders such as '%:p' are processed by the `expand` function provided by vim (see `:h expand`). '%p:h' will return the directory containing the file getting processed. Does that work for your case?

**Comment by ThermalSpan** [comment]
Excellent, thank you for the information. My apologies for the trouble.
